www-apps/radicale: support Py3.7

Hawk777 opened the pull request:
I’m running this and basic functionality appears to work.

epsilon-0 commented on www-apps/radicale/radicale-2.1.11-r2.ebuild:
```diff
@@ -0,0 +1,95 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
```
bump to EAPI 7

Hawk777 replied:
563ebcb0cc1f4207ae53f70ac054ed73e687bbcf

epsilon-0 commented on www-apps/radicale/radicale-2.1.11-r2.ebuild:
```diff
@@ -0,0 +1,95 @@
+
+EAPI="6"
+
+PYTHON_COMPAT=( python3_6 python3_7 )
```
maybe change to a cleaner format like `python3_{6,7}`

optional

Hawk777 replied:
563ebcb0cc1f4207ae53f70ac054ed73e687bbcf; I don’t mind either way (I find the braced format neither easier nor harder to read personally).

epsilon-0 commented on www-apps/radicale/radicale-2.1.11-r2.ebuild:
```diff
@@ -0,0 +1,95 @@
+	dev-python/python-dateutil[${PYTHON_USEDEP}]
+	bcrypt? ( dev-python/passlib[bcrypt,${PYTHON_USEDEP}] )"
+
+S=${WORKDIR}/${MY_P}
```
add quotes

Hawk777 replied:
563ebcb0cc1f4207ae53f70ac054ed73e687bbcf; I thought they weren’t strictly necessary for variable expansions inside other variable assignments, but I prefer being explicit too.

epsilon-0 commented on www-apps/radicale/radicale-2.1.11-r2.ebuild:
```diff
@@ -0,0 +1,95 @@
+}
+
+pkg_setup() {
+	enewgroup ${PN}
```
remove this and create `acct-user` and `acct-group`: https://wiki.gentoo.org/wiki/Categories_acct-group_and_acct-user

Hawk777 replied:
So am I supposed to (1) create acct-user/acct-group packages with myself as maintainer, even though I have nothing to do with the user of those accounts; or (2) create them with maksbotan as their maintainer, thus forcing packages into his maintainership unilaterally? I didn’t know the answer to this question so I didn’t migrate at all, for now, since [this devmanual page](https://devmanual.gentoo.org/ebuild-writing/users-and-groups/index.html) said “The historical way … must not be used for new packages” and this is not a new package, only a revbump. But I can create them if you tell me which maintainer I should put on them.

Hawk777 replied:
Left this alone per your comment below.

epsilon-0 commented:
probably try to get the fixes in and get a go ahead from `pkgcheck scan`: https://packages.gentoo.org/packages/dev-util/pkgcheck

Weuxel commented:
Could we please stop nitpicking? This package blocks the whole update system.

Hawk777 commented:
I don’t mind making the requested changes as soon as @epsilon-0 answers my question.

epsilon-0 commented:
Ah, sorry, this got lost in my mails.  
Really, its up to the devs, I just try to help out here.

Has the maintainer responded?  

@Weuxel not sure what you mean, this package doesn't have any reverse dependencies...

@Hawk777 you are correct, it might be better to stick with the old package convention for this one. Ideally, the maintainer should have responded, unfortunately people are busy.  
Stick to old method for now.

Hawk777 commented:
I think what @Weuxel meant is that running `emerge -uatvDN @world` refuses to upgrade anything until every package on your system is 3.7-compatible. I haven’t heard anything from @maksbotan, but then it also appears they weren’t pinged at all on this bug (I’d kind of have expected a bot to do that, but who knows?). Any opinion, @maksbotan? I’ll give it a bit more time and if no reply I’ll fix the other items but leave user handling alone.

Hawk777 commented:
No comment from @maksbotan after a few days, so I rebased and fixed the mentioned items in 563ebcb0cc1f4207ae53f70ac054ed73e687bbcf; @epsilon-0 back to you.

epsilon-0 commented:
Looks okay to me. Will get one of the devs to also take a look.

juippis commented on www-apps/radicale/radicale-2.1.11-r1.ebuild:
```diff
@@ -1,9 +1,9 @@
+EAPI="7"
 
-PYTHON_COMPAT=( python3_6 )
+PYTHON_COMPAT=( python3_{6,7} )
```
You don't need to revbump when changing PYTHON_COMPAT. We usually require "up-to-standards" ebuilds when you commit a new ebuild (hence @epsilon-0 nitpicking a lot), but for this, no new file is needed. Just edit the PYTHON_COMPAT in place.

Let me knof if you cba to update this, I'll commit it then.

Hawk777 replied:
Oops, it actually already wasn’t a revbump, I just forgot to change the title and commit message. Fixed in 6b0350e0e5e145013d052e509ed9b11b1e2b4e11.

juippis replied:
Sorry but I still see irrelevant changes to this ebuild. **ONLY** edit `PYTHON_COMPAT`, **nothing** else.

Hawk777 replied:
Sorry, I didn’t understand what you said; I thought you meant “you don’t need to revbump”, not “please do not touch anything other than PYTHON_COMPAT”. I thought all the changes I did here were minor cleanups that did not require revbump (EAPI bump, quoting a variable, and irrelevant dep changes) because they do not affect the installed files or, to a material extent, the depgraph. But if that’s what you prefer, then e87191b19479d6babe771ff31078fd5838d202d6.
